inc/targetticket.class.php

```diff
@@ -1069,6 +1089,86 @@ protected function setTargetLocation($data, $formanswer) {
       return $data;
    }
 
+   protected function setTargetContract($data, $formanswer) {
+      global $DB;
+
+      $contract = null;
+      switch ($this->fields['contract_rule']) {
+         case self::CONTRACT_RULE_ANSWER:
+            $contract = $DB->request([
+               'SELECT' => ['answer'],
+               'FROM'   => PluginFormcreatorAnswer::getTable(),
+               'WHERE'  => [
+                  'plugin_formcreator_formanswers_id' => $formanswer->fields['id'],
+                  'plugin_formcreator_questions_id'   => $this->fields['contract_question']
+               ]
+            ])->current();
+            if (isset($contract['answer']) && ctype_digit($contract['answer'])) {
+               $contract = $contract['answer'];
+            }
+            break;
+         case self::CONTRACT_RULE_SPECIFIC:
+            $contract = $this->fields['contract_question'];
+            break;
+         case self::CONTRACT_RULE_LAST_ANSWER:
+            $form_answer_id = $formanswer->fields['id'];
+
+            // Get all answers for dropdown questions of this form, ordered
+            // from last to first displayed
+            $answers = $DB->request([
+               'SELECT' => ['answer.plugin_formcreator_questions_id', 'answer.answer', 'question.values'],
+               'FROM' => PluginFormcreatorAnswer::getTable() . ' AS answer',
+               'JOIN' => [
+                  PluginFormcreatorQuestion::getTable() . ' AS question' => [
+                     'ON' => [
+                        'answer' => 'plugin_formcreator_questions_id',
+                        'question' => 'id',
+                     ]
+                  ]
+               ],
+               'WHERE' => [
+                  'answer.plugin_formcreator_formanswers_id' => $form_answer_id,
+                  'question.fieldtype'                       => "glpiselect",
+               ],
```

Comment:
itemtype in questions table helps you to improbe the SQL query. A 2nd snippet  below will become useless.
```suggestion
               'WHERE' => [
                  'answer.plugin_formcreator_formanswers_id' => $form_answer_id,
                  'question.fieldtype'                       => "glpiselect",
                  'question.itemtype'                       => Contract::class
               ],
```